src/components/utils/file.js

```diff
@@ -1,16 +1,50 @@
 import { findKey } from '~utils/object'
-import { isOfMimeType } from '~utils/blob'
+import { isOfMimeType, canvasToBlob } from '~utils/blob'
 
-const defaultFileTypesAccepted = ['jpg', 'jpeg', 'png', 'pdf']
-const maxFileSizeAcceptedByAPI = 10000000 // The Onfido API only accepts files below 10 MB
+const DEFAULT_ACCEPTED_FILE_TYPES = ['jpg', 'jpeg', 'png', 'pdf']
+const MAX_FILE_SIZE_ACCEPTED_BY_API = 10000000 // The Onfido API only accepts files below 10 MB
 export const validateFileTypeAndSize = (
   file,
-  acceptedTypes = defaultFileTypesAccepted
+  acceptedTypes = DEFAULT_ACCEPTED_FILE_TYPES
 ) =>
   findKey(
     {
       INVALID_TYPE: (file) => !isOfMimeType(acceptedTypes, file),
-      INVALID_SIZE: (file) => file.size > maxFileSizeAcceptedByAPI,
+      INVALID_SIZE: (file) => file.size > MAX_FILE_SIZE_ACCEPTED_BY_API,
     },
     (checkFn) => checkFn(file)
   )
+
+export const resizeImageFile = (file, onImageResize) => {
```

Comment:
What's the impact of this conversion on the performance? Does this operation blocks the UI while running? If so we could consider running it inside a Web Worker or using `FileReaderSync`

Reply:
Now that you mention it, I think it does block the UI for a bit. I'll look into using `FileReaderSync` or the Web Worker solution if that doesn't show much difference.

Reply:
Was having issues with implementing this as a Web Worker solution but from discussion with team doesn't look like the slight delay when it's resizing is too bad/noticeable so not necessary for now.